data/default.nix

```diff
@@ -6,27 +6,47 @@ with (import nixpkgs config);
 
 let
   mkDerivation =
-    { srcs ? ./elm-srcs.nix
-    , src
-    , name
+    { elmSrcs ? ./elm-srcs.nix
+    , srcs
     , srcdir ? "./src"
+    , extension ? ".elm"
+    , name
     , targets ? []
     , versionsDat ? ./versions.dat
     }:
     stdenv.mkDerivation {
-      inherit name src;
+      inherit name srcs;
+      sourceRoot = ".";
```

Comment:
this can be hardcoded - we place generated elm.json with correct paths and the prefix `elm make` with right dirs.